### PR TITLE
[WIP] Add content test jobs for ROSA hcp

### PR DESCRIPTION
**Pull request by xiaojiey**

### CODE REVIEW

**Comment by xiaojiey** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-node-weekly

**Comment by xiaojiey** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-node-weekly

File: ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml [hunk 1]
```diff
@@ -810,6 +810,106 @@ tests:
...
           cpu: 100m
     workflow: ipi-aws
+- as: e2e-aws-rosa-ocp4-pci-dss-node-weekly
+  cron: 0 22 * * 6
+  steps:
+    cluster_profile: quay-aws
+    env:
+      ADDITIONAL_SECURITY_GROUP: "true"
+      CHANNEL_GROUP: nightly
+      OCM_LOGIN_ENV: integration
```
Comment (rhmdnd) [inline ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml:820]: Looks like we need to wire up an OCM token for the quay-aws profile.

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-node-weekly

File: ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml (hunk 1 above)
Comment (rhmdnd) [reply ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml:820]: This should be done now.

**Comment by xiaojiey** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-pci-dss-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-ocp4-cis-node-weekly

File: ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml [hunk 2]
```diff
@@ -810,6 +810,106 @@ tests:
...
+        export CONTENT_IMAGE=ghcr.io/complianceascode/k8scontent:latest
+        export component=ocp4-content-ds
+        git clone https://github.com/ComplianceAsCode/ocp4e2e.git ocp4e2e
+        pushd ocp4e2e
+        go test -v -timeout 120m github.com/ComplianceAsCode/ocp4e2e -profile="$PROFILE" -product="$PRODUCT" -content-image="$CONTENT_IMAGE"
+      from: src
+      resources:
+        requests:
+          cpu: 100m
+    workflow: rosa-aws-sts-hcp
```
Comment (rhmdnd) [inline ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml:837]: I'm struggling to understand how this is wired up to the trace we're getting from CI, which looks like:

```
INFO[2024-04-18T21:11:09Z] Logging into integration with offline token
ERR: Failed to create OCM connection: error creating connection. Not able to get authentication token: access and refresh tokens are unavailable or expired, and there are no password or client secret to request new ones 
```

The [rosa-aws-sts-hcp](https://steps.ci.openshift.org/workflow/rosa-aws-sts-hcp) has several steps but ultimately gets to a [rosa-cluster-provision](https://steps.ci.openshift.org/reference/rosa-cluster-provision) step. The error we should be getting from that script doesn't match what we're observing in CI:

https://steps.ci.openshift.org/reference/rosa-cluster-provision#line102

It looks like we're failing on the [rosa-sts-account-roles-create](https://steps.ci.openshift.org/reference/rosa-sts-account-roles-create#line30) step based on the log and the trace...

But I don't know how that's getting wired up when it's not in the `rosa-aws-sts-hcp` chain?

Comment (rhmdnd) [reply ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml:837]: I needed to implement SSO support in https://github.com/openshift/release/pull/51154 to get this further along with our CI service account.

File: ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml [hunk 3]
```diff
@@ -810,6 +810,106 @@ tests:
...
+- as: e2e-aws-rosa-ocp4-cis-weekly
+  cron: 0 22 * * 6
+  steps:
+    cluster_profile: quay-aws
+    env:
+      ADDITIONAL_SECURITY_GROUP: "true"
+      CHANNEL_GROUP: nightly
+      OCM_LOGIN_ENV: integration
+      OPENSHIFT_VERSION: "4.15"
```
Comment (rhmdnd) [inline ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml:846]: In my testing with https://github.com/openshift/release/pull/50090 I needed to update this to `4.15.8` and change the CHANNEL_GROUP to `stable`.

The nightly channel group seemed to have more specific version strings (4.15.8 wasn't an option).

Comment (xiaojiey) [reply ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml:846]: Correct. I will update to use stable build instead.

**Comment by xiaojiey** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-sts-hcp-fips-ocp4-cis-weekly

**Comment by rhmdnd** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-sts-hcp-fips-ocp4-cis-weekly

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-sts-hcp-fips-ocp4-cis-weekly

**Comment by rhmdnd** [comment]
I think you can rebase this now to pickup https://github.com/openshift/release/pull/51154

**Comment by xiaojiey** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-sts-hcp-fips-ocp4-pci-dss-weekly

File: ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml [hunk 4]
```diff
@@ -810,6 +810,102 @@ tests:
...
+    - as: test-ocp4-cis
+      cli: latest
+      commands: |
+        export PROFILE=cis
+        export PRODUCT=ocp4
+        export CONTENT_IMAGE=ghcr.io/complianceascode/k8scontent:latest
+        export component=ocp4-content-ds
+        git clone https://github.com/ComplianceAsCode/ocp4e2e.git ocp4e2e
+        pushd ocp4e2e
+        go test -v -timeout 120m github.com/ComplianceAsCode/ocp4e2e -profile="$PROFILE" -product="$PRODUCT" -content-image="$CONTENT_IMAGE"
```
Comment (Vincent056) [inline ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml:855]:
```
go test -v -timeout 120m github.com/ComplianceAsCode/ocp4e2e -profile="$PROFILE" -product="$PRODUCT" -content-image="$CONTENT_IMAGE" -platform="rosa"
```
You can use this to run the test

**Comment by xiaojiey** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-sts-hcp-fips-ocp4-pci-dss-node-weekly

File: ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml [hunk 5]
```diff
@@ -810,6 +810,102 @@ tests:
         requests:
           cpu: 100m
     workflow: ipi-aws
+- as: e2e-aws-rosa-sts-hcp-fips-ocp4-pci-dss-node-weekly
```
Comment (rhmdnd) [inline ci-operator/config/ComplianceAsCode/content/ComplianceAsCode-content-master__4.15.yaml:813]: We might need to add:

```yaml
base_images:
  cli-ocm:
    name: cli-ocm
    namespace: ci
    tag: latest
  rosa-aws-cli:
    name: rosa-aws-cli
    namespace: ci
    tag: release
```

To get past some of the CI issues pulling those images:

```
* 2024-05-16T07:32:06Z 261x kubelet: Back-off pulling image "image-registry.openshift-image-registry.svc:5000/ci-op-jcgs5ks6/stable:rosa-aws-cli"
```

I needed to do something similar with the compliance-operator job.

https://github.com/openshift/release/blob/master/ci-operator/config/ComplianceAsCode/compliance-operator/ComplianceAsCode-compliance-operator-master.yaml#L6-L17

**Comment by rhmdnd** [comment]
This should help get the operator running on the worker nodes: https://github.com/ComplianceAsCode/ocp4e2e/pull/41

**Comment by xiaojiey** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-sts-hcp-fips-ocp4-pci-dss-node-weekly

**Comment by xiaojiey** [comment]
/pj-rehearse periodic-ci-ComplianceAsCode-content-master-4.15-e2e-aws-rosa-sts-hcp-fips-ocp4-pci-dss-node-weekly

**Pull request closed without merging**